packages/flutter/test/widgets/image_test.dart

```diff
@@ -35,6 +35,57 @@ void main() {
     imageCache.maximumSize = originalCacheSize;
   });
 
+  testWidgets('Verify Image does not use disposed handles', (WidgetTester tester) async {
+    final ValueNotifier<int> outerListenable = ValueNotifier<int>(0);
+    final ValueNotifier<int> innerListenable = ValueNotifier<int>(0);
+
+    bool imageLoaded = false;
+    MemoryImage image = MemoryImage(Uint8List.fromList(kTransparentImage));
+
+    Future<void> runAsyncAndIdle() async {
+      for (int i = 0; i < 20; ++i) {
+        await tester.runAsync(() => Future<void>.delayed(Duration.zero));
+        await tester.idle();
+      }
+    }
```

Comment:
This is a little difficult to reason about and may cause the test to take longer than needed or in fact never terminate (for example, if the image cache manages to get a hold of a completer that was created in the fake async zone, no amount of real async delays will fix that). I suspect we're getting lucky right now and not running into this case, but we could.

I think the easiest option would be to use `_TestImageProvider`s which you can manually complete with real-async images (like the `image10x10` defined above in this test file). Alternatively, you can use a memory image but you have to be more careful to make sure the image cache doesn't get involved and more explicit about what we're waiting for here -and asserting that the thing we're waiting for actually happens.